tests/unit/docs/test_params.py

```diff
@@ -0,0 +1,262 @@
+        self.include_params = [
+            DocumentedShape(
+                name='Baz', type_name='integer',
+                documentation='This describes baz.'
+            )
+        ]
+
+    def test_request_params(self):
+        self.request_params.document_params(
+            self.doc_structure, self.operation_model.input_shape,
+            include=self.include_params
+        )
+        self.assert_contains_lines([
+            ':type Foo: string',
+            ':param Foo: This describes foo.',
+            ':type Baz: int',
+            ':param Baz: This describes baz.'
+        ])
+
+    def test_response_params(self):
+        self.response_params.document_params(
+            self.doc_structure, self.operation_model.input_shape,
+            include=self.include_params
+        )
+        self.assert_contains_lines([
+            '- *(dict) --*',
+            '  - **Foo** *(string) --* This describes foo.',
+            '  - **Baz** *(integer) --* This describes baz.'
+        ])
+
+
+class TestDocumentExclude(BaseParamsDocumenterTest):
+    def setUp(self):
+        super(TestDocumentExclude, self).setUp()
+        self.add_shape_to_params('Foo', 'String', 'This describes foo.')
+        self.add_shape_to_params('Bar', 'String', 'This describes bar.',
+                                 is_required=True)
+        self.exclude_params = ['Foo']
+
+    def test_request_params(self):
+        self.request_params.document_params(
+            self.doc_structure, self.operation_model.input_shape,
+            exclude=self.exclude_params)
+        self.assert_contains_lines([
+            ':type Bar: string',
+            ':param Bar: **[REQUIRED]** This describes bar.'
+        ])
+        self.assert_not_contains_lines([
+            ':type Foo: string',
+            ':param Foo: This describes foo.'
+        ])
+
+    def test_response_params(self):
+        self.response_params.document_params(
+            self.doc_structure, self.operation_model.input_shape,
+            exclude=self.exclude_params)
+        self.assert_contains_lines([
+            '- *(dict) --*',
+            '  - **Bar** *(string) --* This describes bar.'
+        ])
+        self.assert_not_contains_line(
+            '  - **Foo** *(string) --* This describes foo.')
+
+
+class TestDocumentList(BaseParamsDocumenterTest):
+    def setUp(self):
+        super(TestDocumentList, self).setUp()
+        self.add_shape(
+            {'List': {
+                'type': 'list',
+                'member': {'shape': 'String',
+                           'documentation': 'A string element'}}})
+        self.add_shape_to_params('Foo', 'List', 'This describes the list.')
+
+    def test_request_params(self):
+        self.request_params.document_params(
+            self.doc_structure, self.operation_model.input_shape)
+        self.assert_contains_lines([
+            ':type Foo: list',
+            ':param Foo: This describes the list.',
```

Comment:
Perhaps we could add some kind of transition text here.  I found this a bit hard to follow in the generated docs.  Something like "Each element of this list is a... <docs>".